src/lib/Select.svelte

```diff
@@ -1,90 +1,316 @@
+            onclear(value);
+            value = undefined;
+            closeList();
+            handleFocus();
+        },
+        oninput = () => {},
+        onfilter = () => {},
+        onclear = () => {},
+        onfocus = () => {},
+        onchange = () => {},
+        onselect = () => {},
+        onblur = () => {},
+        onhoveritem = () => {},
+        onerror = () => {},
+        onloaded = () => {},
+        ...rest
+    } = $props();
+
+    let list = $state();
+    let listDom = $state();
+    let _inputAttributes = $state();
+    let activeValue = $state(undefined);
+    let prev_value = $state();
+    let prev_filterText = $state();
+    let prev_multiple = $state();
+    let isScrollingTimer = $state();
+    let isScrolling = $state(false);
+    let prefloat = $state(true);
+    let hasValue = $state(false);
+    let placeholderText = $state('');
+    let showClear = $state(false);
+    let hideSelectedItem = $state(false);
+    let ariaSelection = $state('');
+    let filteredItems = $state(undefined);
+    let ariaContext = $state('');
+    let scrollToHoverItem = $state(0);
+    // let scrollToHoverItem = $derived(hoverItemIndex);
+    let _floatingConfig = $state({
+        strategy: 'absolute',
+        placement: 'bottom-start',
+        middleware: [offset(listOffset), flip(), shift()],
+        autoUpdate: false,
+    });
+
+    const activeScroll = scrollAction;
+    const hoverScroll = scrollAction;
+    const [floatingRef, floatingContent, floatingUpdate] = createFloatingActions(_floatingConfig);
+
+    onMount(() => {
+        if (listOpen) focused = true;
+        if (focused && input) input.focus();
+    });
+    $effect.pre(() => {
+        placeholderText =
+            placeholderAlwaysShow && multiple
+                ? placeholder
+                : multiple && value?.length === 0
+                  ? placeholder
+                  : value
+                    ? ''
+                    : placeholder;
+    });
+    $effect.pre(() => {
+        filterText;
+        value;
+        items;
+        untrack(
+            () =>
+                (filteredItems = filter({
+                    loadOptions,
+                    filterText,
+                    items,
+                    multiple,
+                    value,
+                    itemId,
+                    groupBy,
+                    label,
+                    filterSelectedItems,
+                    itemFilter,
+                    convertStringItemsToObjects,
+                    filterGroupedItems,
+                })),
+        );
+    });
+    $effect.pre(() => {
+        prev_multiple = multiple;
+    });
+    $effect(() => {
+        hasValue;
+        untrack(() => {
+            if (items) setValue(items);
+        });
+    });
+    $effect(() => {
+        searchable;
+        untrack(() => {
+            if (inputAttributes || !searchable) assignInputAttributes();
+        });
+    });
+    $effect(() => {
```

Comment:
```suggestion
    onMount(() => {
```

An effect that doesn't track anything is just going to run once, when mounting.